### PR TITLE
Fix registry/attribute errors in tests mocking the server player

### PR DESCRIPTION
**Pull request by Steveplays28**
- [x] Mock `ServerPlayer#connection`
- [x] Fix region ticket verification
- [x] Re-enable `TestServerPlayer#testTeleportToVanilla`

### CODE REVIEW

File: src/test/java/io/github/opencubicchunks/cubicchunks/mixin/test/common/server/MixinServerPlayerTest.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import net.minecraft.server.network.ServerGamePacketListenerImpl;
+import org.mockito.Mockito;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+
+@Mixin(ServerPlayer.class)
+public abstract class MixinServerPlayerTest {
+    @Shadow
+    @SuppressWarnings("unused")
+    public ServerGamePacketListenerImpl connection = Mockito.mock();
```
Comment (NotStirred) [inline src/test/java/io/github/opencubicchunks/cubicchunks/mixin/test/common/server/MixinServerPlayerTest.java]: Is this mixin necessary? Can we instead assign the connection in `TestServerPlayer#setupServerPlayer`?

**Review by NotStirred — COMMENTED** [review]
Minor comments
Thoughts on having a TODO comment on each of the neoforge registry fixes? I think they're only necessary due to us using loom for running tests while neoforge gets junit working

File: src/test/java/io/github/opencubicchunks/cubicchunks/mixin/test/common/server/MixinServerPlayerTest.java (hunk 1 above)
Comment (Steveplays28) [reply src/test/java/io/github/opencubicchunks/cubicchunks/mixin/test/common/server/MixinServerPlayerTest.java]: Yep, that'd be better, didn't realize it was `public`. I'll change that.

**Comment by Steveplays28** [comment]
@NotStirred I've added TODO comments to all the mixins:
```java
// TODO: Remove this mixin class when NeoForge supports JUnit for tests
//  This mixin class is only required due to NeoForge not supporting JUnit for testing yet (a workaround is currently used, involving Loom)
...
```

**Review by NotStirred — APPROVED** [review]

**Pull request merged**